packages/browser-wallet/src/popup/pages/Account/TransactionLog/TransactionList.tsx

```diff
@@ -226,9 +246,12 @@ export default function TransactionList({ onTransactionClick }: TransactionListP
         getTransactions(address, transactionResultLimit, 'descending', fromId)
             .then((transactionResult) => {
                 setHasNextPage(transactionResult.full);
+                removePendingTransactions(transactionResult.transactions);
```

Comment:
A pending transaction is never removed if you don't get the transaction externally. What could happen is that you send a transaction and close your browser / or just switch to a different account. Before returning to your wallet/account you receive a lot of transactions, more than `transactionResultLimit`, and then the pending transaction will still be visible at the top. The user then has to scroll through their transactions for the pending one to be removed.

I believe that the desktop wallet solves this by listening for the status of outgoing transactions, and does not rely on it coming back from the wallet proxy.

Reply:
Yes, good catch, I'll revise.